app/api/services/api_key.py

```diff
@@ -0,0 +1,45 @@
+from app.api.helpers import Service
+from app.models import ApiKey, db
+from app.api.helpers import generate_random_token
+from datetime import datetime
+
+
+class ApiKeyService(Service):
+    __model__ = ApiKey
+
+    def __init__(self, *args, **kwargs):
+        super(ApiKeyService, self).__init__(*args, **kwargs)
+
+    def get_key(self, key):
+        query = (
+            db
+            .session
+            .query(ApiKey)
+            .filter(
+                ApiKey.key == key,
+                ApiKey.revoked_at.is_(None)
+            )
+            .order_by(ApiKey.created_at.desc())
+        )
+        return query.first()
+
+    def generate(self, user_id, length=32):
+        key = ApiKey(user_id=user_id, key=generate_random_token(length=length))
```

Comment:
Should we fail to generate the key if `length` is below 32?

Reply:
The DB constraint and max column length should combine to prevent anything except a 64 character (32 bytes hex encoded) string being stored at the DB level.